### ISSUE TITLE
Ensure that calls to `reflect` methods and functions are safe

### ISSUE THREAD

**Issue by zapateo** [krenalis/krenalis#261]
Ensure that calls to 'reflect' methods and functions are safe. In some circumstances they may fail with a panic, opening security issues.

We should investigate on this.

**Comment by zapateo** [comment]
@gazerro, regarding this issue (very old), I don't see how to approach it to solve it, if not by doing a general review of all the Meergo code.

I would be in favor of closing it, since it doesn't seem to add anything particular. What do you think about it?

**Comment by gazerro** [comment]
@zapateo Yes, It’s like saying we need to make sure all slice accesses are within bounds and don’t go out of range. I don’t think there’s any part of the code where this hasn’t been considered. The issue was probably raised after just one confirmed case.
